backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/HttpRequestDetails.java

```diff
@@ -0,0 +1,95 @@
+package gov.cdc.usds.simplereport.db.model.auxiliary;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The information that we capture about the HTTP request for an audit event. Must be stable (no
+ * type changes or field deletions) once released, since we will be using it to serialize records.
+ * Mostly we should expect the {{@link #getForwardedAddresses()} information to be what anybody ever
+ * consults, but the other stuff is easy to save here and hard to reconstruct otherwise.
+ */
+public class HttpRequestDetails {
+
+  /** Headers we extract from the request. */
+  private static final class Headers {
+    private static final String FORWARDED_PROTOCOL = "x-forwarded-proto";
+    private static final String FORWARDED_CLIENT = "x-forwarded-for";
+    private static final String FORWARDED_HOST = "x-original-host"; // screw you, Azure
```

Comment:
Maybe a more helpful comment as to why `x-original-host` is required? 🤪

Reply:
I mean, it's because Azure app services doesn't use `x-forwarded-host` like everybody else in the known world—I'm not sure there's much more to say than I already said. 😝